Objects/longobject.c

```diff
@@ -415,6 +415,18 @@ PyLong_FromUnsignedLong(unsigned long ival)
 PyObject *
 PyLong_FromDouble(double dval)
 {
+    /* Try to get out cheap if this fits in a long.  The attempt
+     * to cast to long must be protected, as C doesn't define what
+     * happens if the double is too big to fit in a long.  Some rare
+     * systems raise an exception then (RISCOS was mentioned as one,
+     * and someone using a non-default option on Sun also bumped into
+     * that).
+     */
+    const double int_max = pow(2, CHAR_BIT * sizeof(long) - 1);
```

Comment:
Are there guaranties that it is evaluated at compile time and not every time when call `PyLong_FromDouble()`? Why not use `LONG_MAX`?

Reply:
Fixed.